### PR TITLE
AG-11993 Add zoom to initialState

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-11993

This also fixes some issues with independent axis zooming (see ticket for details).

### CODE REVIEW

**Review by alantreadway — APPROVED** [review]
LGTM

**Comment by lsjroberts** [comment]
Leaving as draft until we are ready to put this into a release.

File: packages/ag-charts-types/src/chart/chartBuilderOptions.ts [hunk 1]
```diff
@@ -137,9 +137,15 @@ export interface ImageDataUrlOptions {
     fileFormat?: string;
 }
 
+// TODO: this is a breaking change as previously it was permissive of any type for each key
+// export interface AgChartState extends AgInitialStateOptions {
+//     version: string;
+// }
+
```
Comment (lsjroberts) [inline packages/ag-charts-types/src/chart/chartBuilderOptions.ts]: We should have been doing this previously, but can't introduce it just yet as it is a breaking change.

File: packages/ag-charts-community/src/sparklines-util.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export * from './util/vector';
 export * from './util/zip';
 
 export { isValidDate as isDate, isFiniteNumber as isNumber, isString } from './util/type-guards';
-export { bifurcate, extent, normalisedExtent, normalisedExtentWithMetadata } from './util/array';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/sparklines-util.ts:15]: Can you confirm this isn't being used in the Grid repo?

Comment (lsjroberts) [reply packages/ag-charts-community/src/sparklines-util.ts:15]: Yep, not used (I introduced it specifically for this use-case) - https://github.com/search?q=repo%3Aag-grid%2Fag-grid%20birfurcate&type=code

File: packages/ag-charts-types/src/chart/chartBuilderOptions.ts (hunk 1 above)
Comment (lsjroberts) [reply packages/ag-charts-types/src/chart/chartBuilderOptions.ts]: Added deprecation notice

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**